iXBRLViewerPlugin/viewer/src/js/viewer.js

```diff
@@ -486,7 +531,13 @@ Viewer.prototype.selectElementByClick = function (e) {
     var itemIDList = [];
     var viewer = this;
     var sameContentAncestorId;
-    e.parents(".ixbrl-element").addBack().each(function () { 
+    // If the user clicked on a sub-element, treat as if we clicked the first
+    // non-sub-element ancestor in the DOM hierarchy - which would typically be
+    // the corresponding ixbrl-element.
+    if (e.hasClass('ixbrl-sub-element')) {
+        e = e.parents('.ixbrl-element:not(.ixbrl-sub-element)').first();
```

Comment:
Out of curiosity, what happens for nested facts?  It seems like it would be possible for the nested fact region to have both ixbrl-sub-element and ixbrl-element, and then it wouldn't be possible to click on the nested element.  If it's not a problem, can you expand the comment to explain why this logic doesn't cause that problem?

Reply:
Additional comments added.

The subtlety here is that `.ixbrl-sub-element` gets applied directly to elements in the source document, whereas `.ixbrl-element` gets added to wrapper nodes that are inserted by the viewer, so we can ignore `.ixbrl-sub-element` elements without missing a real tag.  

Note that this particular line is only needed for the "same content ancestor" detection in the following traversal.

There is one possible edge case: for tags in table cells we sometimes use the enclosing `<td>` or `<th>` as the `ixbrl-element` rather than using a wrapper.  If the table cell were to be absolutely positioned, and to be within another text tag, then we'd have an element that would be both an `ixbrl-element` and an `ixbrl-sub-element`.  In this case I think we would miss the nested table cell tag.  This seems unlikely, although PDF-converted HTML never ceases to surprise :) 

I think we could address it by inverting the tests and checking for the "real" element classes (`.ixbrl-element-nonfraction` etc.) rather than the absence of `.ixbrl-sub-element`.

Reply:
> If the table cell were to be absolutely positioned, and to be within another text tag, then we'd have an element that would be both an ixbrl-element and an ixbrl-sub-element. In this case I think we would miss the nested table cell tag. This seems unlikely

This scenario seems plausible for a text block surrounding a table.  I believe Workiva currently uses `<td><div absolutely-positioned-style>`, but we have considered switching to `<td absolutely-positioned-style>`.  I think we should try to not break that use-case if possible.

Reply:
> I think we could address it by inverting the tests and checking for the "real" element classes (.ixbrl-element-nonfraction etc.) rather than the absence of .ixbrl-sub-element.

Would that cause problems for continuations in tables nested within text blocks?  Perhaps that's unlikely enough that we can use that approach and just add a comment that it might cause problems.

Reply:
> Would that cause problems for continuations in tables nested within text blocks?

I think that should be OK, as we can include `.ixbrl-continuation` in the list of real element classes.

The problem with the approach is that we can end up with the same fact twice in `itemIDList` and thus appearing twice in the inspector, as we'll get the same ID from the real `.ixbrl-element` and the `.ixbrl-sub-element`, so we need to deduplicate `itemIDList`.

I'll create some test cases for this, and then attempt to fix it.

Reply:
This is done in 74a99a557cacb1380ae3d04893a61ad5e55ff212.

I've realised that there's actually no need to store IDs in `ivid` on sub-elements, as we always get the list of IDs from the ancestor `.ixbrl-element`s anyway.  This removes the need to de-duplicate.

I've also reworked the classes, so that sub elements are no longer also given the `ixbrl-element` class (unless they are also `ixbrl-element` by virtue of being a `<td>` or `<th>` acting as a wrapper).